packages/filesystem/src/node/download/file-download-handler.ts

```diff
@@ -43,46 +42,98 @@ export abstract class FileDownloadHandler {
+                    response.setHeader('Content-Range', `bytes */${statSize}`);
+                    // Return the 416 'Requested Range Not Satisfiable'.
+                    response.status(REQUESTED_RANGE_NOT_SATISFIABLE).end();
+                    return;
+                }
+                response.setHeader('Content-Range', `bytes ${rangeStart}-${rangeEnd}/${statSize}`);
+                response.setHeader('Content-Length', rangeStart === rangeEnd ? 0 : (rangeEnd - rangeStart + 1));
+                response.setHeader('Cache-Control', 'no-cache');
+                this.streamDownload(PARTIAL_CONTENT, response, fs.createReadStream(filePath, { start: rangeStart, end: rangeEnd }), id);
+            }
+        } catch (e) {
+            this.fileDownloadCollection.deleteDownload(id);
+            this.handleError(response, e, INTERNAL_SERVER_ERROR);
+        }
+    }
+    protected streamDownload(status: number, response: Response, stream: fs.ReadStream, id: string) {
+        response.status(status);
+        stream.on('error', error => {
+            this.fileDownloadCollection.deleteDownload(id);
+            this.handleError(response, error, INTERNAL_SERVER_ERROR);
+        });
+        response.on('error', error => {
+            this.fileDownloadCollection.deleteDownload(id);
+            this.handleError(response, error, INTERNAL_SERVER_ERROR);
+        });
+        response.on('close', () => {
+            stream.destroy();
+        });
+        stream.pipe(response);
+    }
+    protected parseRangeHeader(range: string | string[] | undefined, statSize: number): { start: number, end: number } | undefined {
+        if (!range || range.length === 0 || Array.isArray(range)) {
+            return;
+        }
+        const index = range.indexOf('=');
+        if (index === -1) {
+            return;
+        }
+        const rangeType = range.slice(0, index);
+        if (rangeType !== 'bytes') {
+            return;
+        }
+        const [start, end] = range.slice(index + 1).split('-').map(r => parseInt(r, 10));
+        return {
+            start: isNaN(start) ? 0 : start,
+            end: (isNaN(end) || end > statSize - 1) ? (statSize - 1) : end
+        };
+    }
     protected async archive(inputPath: string, outputPath: string = path.join(os.tmpdir(), v4()), entries?: string[]): Promise<string> {
         await this.directoryArchiver.archive(inputPath, outputPath, entries);
         return outputPath;
     }
 
-    protected async deleteRecursively(pathToDelete: string): Promise<void> {
-        rimraf(pathToDelete, error => {
-            if (error) {
-                this.logger.warn(`An error occurred while deleting the temporary data from the disk. Cannot clean up: ${pathToDelete}.`, error);
-            }
-        });
-    }
-
-    protected async createTempDir(): Promise<string> {
-        const outputPath = path.join(os.tmpdir(), v4());
+    protected async createTempDir(downloadId: string): Promise<string> {
```

Comment:
Please keep the existing API, if possible. It should be:
`protected async createTempDir(downloadId: string = v4()): Promise<string> {`

Reply:
this wasn't even used anywhere before. But yeah ill add a default.